mono/io-layer/wthreads.c

```diff
@@ -358,7 +352,11 @@ GetThreadPriority (gpointer handle)
 				  (gpointer *)&thread_handle);
 				  
 	if (ok == FALSE) {
-		return (THREAD_PRIORITY_NORMAL);
+		pthread_getschedparam (pthread_self(), &policy, &param);
+		if ((policy == SCHED_FIFO) || (policy == SCHED_RR)) 
+			return (((MonoInternalThread *)handle)->priority);
```

Comment:
A `HANDLE` is an opaque structure, it's not a pointer to the managed thread object.
(If I've chased the code down correctly, I'm reasonably sure it's some kind of pthread index on posix platforms).